### DISCUSSION TITLE
What's your workflow?

### DISCUSSION THREAD

**Discussion by jj-uk** [wxFormBuilder/wxFormBuilder#866] [Q&A]
I'm using this tool for the 1st time- I'm also using Visual Studio to compile/run/debug.

In FB, i'm designing the GUI, but what's the workflow to actually run-modify-run the generated code? The tool has no compiler option, no code editor.

How are you using the output to create your app? Copy/pasting sections of the output into VisualStudio, running it, adding/modifying in FB, and repeating? This seems tedious, but as it's writing large amounts of code for me, I guess that's ok.

I looked at code::blocks as an all-in-one, but it seems to be discontinued, so i'd like to continue with FB, but I just need to get my head around the workflow.

**Comment by sodevel** [comment]
wxFB does generate source code from your design which then has to be processed by your build system.

You have basically two different output options, scource code for your desired programming language or XRC. XRC is a description language of the wxWidgets framework that allows to load component at runtime. This output is inferiour to generated code because not everything that can be modelled inside wxFB can be exported as XRC. Also, there is more extra work necessary in your application to e.g. bind event handlers and do data binding.

The generated source code actually consists of two parts. The first part is the generated design of wxFB, these classes must not be used directly, they get rewritten every time you generate the code. The second part is the implementation that actually derives from the generated classes. This can also be created from wxFB and these files are supposed to be edited by you to perform data binding, event handler implementation or any other custom logic. Although wxFB offers to update these files, i do not recommend to do so, this feature is kind of fragile, also, the only things that get updated are new empty event handler implementations, this can be done manually by copying the corresponding lines from the generated code.

You add both of these generated file sets to your build system so that they get compiled by it.
